src/gtest/test_transaction_builder.cpp

```diff
@@ -248,12 +255,60 @@ TEST(TransactionBuilder, SproutToSproutAndSapling) {
     RegtestDeactivateSapling();
 }
 
+TEST(TransactionBuilder, TransparentToOrchard)
+{
+    auto consensusParams = RegtestActivateNU5();
+
+    CBasicKeyStore keystore;
+    CKey tsk = AddTestCKeyToKeyStore(keystore);
+    auto scriptPubKey = GetScriptForDestination(tsk.GetPubKey().GetID());
+
+    auto coinType = Params().BIP44CoinType();
+    auto seed = MnemonicSeed::Random(coinType);
+    auto sk = libzcash::OrchardSpendingKey::ForAccount(seed, coinType, 0);
+    auto fvk = sk.ToFullViewingKey();
+    auto ivk = fvk.ToIncomingViewingKey();
+    libzcash::diversifier_index_t j(0);
```

Comment:
```suggestion
    libzcash::diversifier_index_t j(42);
```
(non-zero to check that we are not making assumptions that a default address is used).